test/common/utils_assert.h

```diff
@@ -70,6 +73,23 @@ void AssertSameType( const T& /*x*/, const T& /*y*/ ) {}
         utils::CheckAssertionFailure(__LINE__, #x, okay, message, substr); \
         utils::ResetAssertionHandler(default_handler);                     \
     }
+#else
+#define TEST_CUSTOM_ASSERTION_HANDLER(x, substr)                           \
+    {                                                                      \
+        auto default_handler = utils::SetCustomAssertionHandler();         \
+        const char* message = nullptr;                                     \
+        bool okay = false;                                                 \
+        if (setjmp(utils::g_assertion_jmp_buf)) {                          \
+            okay = true;                                                   \
+            message = utils::g_assertion_failure->message;                 \
+        } else {                                                           \
+            x;                                                             \
+        }                                                                  \
```

Comment:
Does it make sense to swap if and else branches to better match the original test?
I also thought about keeping only this implementation as the only test:)

Reply:
Going one step further, I had an idea to merge two implementations as several lines are identical. However, it might negatively impact the overall readability. Although the macro will be smaller.

Reply:
I think the exception implementation is better overall and should be used when possible.

>  I had an idea to merge two implementations as several lines are identical. However, it might negatively impact the overall readability. Although the macro will be smaller.

This macro is relatively small, so small code duplication is OK here.

I swapped branches.